docs/en/migration/hook.md

```diff
@@ -1,3 +1,319 @@
+    <td>before test</td>
+    <td>None</td>
+    <td>before_test</td>
+  </tr>
+  <tr>
+    <td>after test</td>
+    <td>None</td>
+    <td>after_test</td>
+  </tr>
+  <tr>
+    <td>before each epoch</td>
+    <td>None</td>
+    <td>before_test_epoch</td>
+  </tr>
+  <tr>
+    <td>after each epoch</td>
+    <td>None</td>
+    <td>after_test_epoch</td>
+  </tr>
+  <tr>
+    <td>before each iteration</td>
+    <td>None</td>
+    <td>before_test_iter，with additional args: batch_idx and data_batch</td>
+  </tr>
+  <tr>
+    <td>after each iteration</td>
+    <td>None</td>
+    <td>after_test_iter，with additional args: batch_idx、data_batch and outputs</td>
+  </tr>
+</tbody>
+</table>
+
+## Usage Comparison
+
+In MMCV, to register hooks to the Runner, you need to call the `register_training_hooks` method of the Runner to register hooks on the Runner, while in MMEngine, you can pass parameters to the Runner's initialization method for registration.
+
+- MMCV
+
+```python
+model = ResNet18()
+optimizer = optim.SGD(model.parameters(), lr=0.001, momentum=0.9)
+lr_config = dict(policy='step', step=[2, 3])
+optimizer_config = dict(grad_clip=None)
+checkpoint_config = dict(interval=5)
+log_config = dict(interval=100, hooks=[dict(type='TextLoggerHook')])
+custom_hooks = [dict(type='NumClassCheckHook')]
+runner = EpochBasedRunner(
+    model=model,
+    optimizer=optimizer,
+    work_dir='./work_dir',
+    max_epochs=3,
+    xxx,
+)
+runner.register_training_hooks(
+    lr_config=lr_config,
+    optimizer_config=optimizer_config,
+    checkpoint_config=checkpoint_config,
+    log_config=log_config,
+    custom_hooks_config=custom_hooks,
+)
+runner.run([trainloader], [('train', 1)])
+```
+
+- MMEngine
+
+```python
+model=ResNet18()
+optim_wrapper=dict(
+    type='OptimizerWrapper',
+    optimizer=dict(type='SGD', lr=0.001, momentum=0.9))
+param_scheduler = dict(type='MultiStepLR', milestones=[2, 3]),
+default_hooks = dict(
+    logger=dict(type='LoggerHook'),
+    param_scheduler=dict(type='ParamSchedulerHook'),
+    checkpoint=dict(type='CheckpointHook', interval=5),
+)
+custom_hooks = [dict(type='NumClassCheckHook')]
+runner = Runner(
+    model=model,
+    work_dir='./work_dir',
+    optim_wrapper=optim_wrapper,
+    param_scheduler=param_scheduler,
+    train_cfg=dict(by_epoch=True, max_epochs=3),
+    default_hooks=default_hooks,
+    custom_hooks=custom_hooks,
+    xxx,
+)
+runner.train()
+```
+
+For more details of MMEngine hooks, please refer to [Usage of Hooks](../tutorials/hook.md).
+
+## Implementation Comparison
+
+Taking `CheckpointHook` as an example, compared with [CheckpointHook](https://github.com/open-mmlab/mmcv/blob/v1.6.0/mmcv/runner/hooks/checkpoint.py) in MMCV, [CheckpointHook](https://github.com/open-mmlab/mmengine/blob/main/mmengine/hooks/checkpoint_hook.py) of MMEngine needs to implement the `after_val_epoch` method, since new CheckpointHook supports saving the optimal weights, while in MMCV, the function is achieved by EvalHook.
```

Comment:
```suggestion
Taking `CheckpointHook` as an example, compared with [CheckpointHook](https://github.com/open-mmlab/mmcv/blob/v1.6.0/mmcv/runner/hooks/checkpoint.py) in MMCV, [CheckpointHook](https://github.com/open-mmlab/mmengine/blob/main/mmengine/hooks/checkpoint_hook.py) of MMEngine needs to implement the `after_val_epoch` method, since new `CheckpointHook` supports saving the optimal weights, while in MMCV, the function is achieved by EvalHook.
```